src/components/StatusPageMultiple.js

```diff
@@ -0,0 +1,156 @@
+import { parseOfferId } from '@/utils/parseOfferId';
+import { Alert } from '@/ui/Alert';
+import { Page } from '@/ui/Page';
+import { SelectionTable } from '@/ui/SelectionTable';
+import { Stack } from '@/ui/Stack';
+import { Text } from '@/ui/Text';
+import { getValueFromTheme } from '@/ui/theme';
+import { StatusModal } from './StatusModal';
+import { QueryStatus } from '@/hooks/api/authenticated-query';
+import { camelCase } from 'lodash';
+import { Icons } from '@/ui/Icon';
+import { Button, ButtonVariants } from '@/ui/Button';
+import { Link, LinkVariants } from '@/ui/Link';
+import { OfferStatus } from '@/constants/OfferStatus';
+
+const getValue = getValueFromTheme('statusPage');
+
+const Status = ({ type, reason }) => {
+  const { i18n } = useTranslation();
+  return (
+    <Stack>
+      <Text>{type}</Text>
+      {!!reason?.[i18n.language] && (
+        <Text color={getValue('infoTextColor')}>{reason[i18n.language]}</Text>
+      )}
+    </Stack>
+  );
+};
+
+Status.propTypes = {
+  type: PropTypes.string.isRequired,
+  reason: PropTypes.object,
+};
+
+const StatusPageMultiple = ({ event, refetchEvent }) => {
+  const { t, i18n } = useTranslation();
+
+  const eventId = parseOfferId(event['@id']);
+  const name =
+    event?.name?.[i18n.language] ?? event?.name?.[event.mainLanguage];
+  const subEvents = event?.subEvent;
+
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [selectedRows, setSelectedRows] = useState([]);
+
+  const selectedSubEventIds = useMemo(() => selectedRows.map((row) => row.id), [
+    selectedRows,
+  ]);
+
+  const handleSuccess = async () => {
+    await refetchEvent();
+    setIsModalVisible(false);
+  };
+
+  const {
+    mutate: changeStatusSubEvents,
+    ...changeSubEventsMutation
+  } = useChangeStatusSubEvents({
+    onSuccess: handleSuccess,
+  });
+
+  const handleConfirmChangeStatus = async (type, reason) => {
+    await changeStatusSubEvents({
+      eventId,
+      subEventIds: selectedSubEventIds,
+      subEvents,
+      type,
+      reason:
+        reason.length > 0 && type !== OfferStatus.AVAILABLE
+          ? {
+              [i18n.language]: reason,
+            }
+          : undefined,
+    });
+  };
+
+  const columns = useMemo(
+    () => [
+      {
+        Header: t('offerStatus.period'),
+        accessor: 'period',
+      },
+      {
+        Header: t('offerStatus.statusLabel'),
+        accessor: 'status',
+      },
+    ],
+    [],
+  );
+
+  const data = useMemo(
+    () =>
+      subEvents.map((subEvent) => ({
+        period: formatPeriod(
+          subEvent.startDate,
+          subEvent.endDate,
+          i18n.language,
+          t,
+        ),
+        status: (
+          <Status
+            type={t(
+              `offerStatus.status.event.${camelCase(subEvent.status.type)}`,
+            )}
+            reason={subEvent.status.reason}
+          />
+        ),
+      })),
+    [subEvents],
+  );
+
+  return [
```

Comment:
Is this square brackets syntax new? I always remember it being `return (...);` when returning JSX.

Reply:
The square bracket syntax is needed when we want to apply margins passed from the parent component. Because if we use a React Fragment to wrap these, properties like spacing={5} on the parent component won't work

Reply:
Okay, weird but good to know. Thanks for the clarification! 👍